File: Count the Substrings[ 29-March-2023]/Approach 1[ O(n^3 )]/Solution.cpp
//		>> C++ CODE <<

int countSubstring(string S) {
        
        int ans = 0;
            for (int i = 0; i < S.length(); i++) {
                for (int j = i + 1; j < S.length(); j++) {
                    int ucase = 0, lcase = 0;
                    for (int k = i; k <= j; k++) {
                        if (isupper(S[k])) {
                            ucase++;
                        }
                        else {
                            lcase++;
                        }
                    }
                    if (ucase == lcase) {
                        ans++;
                    }
                }
            }
            return ans;
    }

File: Count the Substrings[ 29-March-2023]/Approach 1[ O(n^3 )]/Solution.java
//		>> JAVA CODE <<

int countSubstring(String S) 
    { 
        int ans=0;
        for(int i=0;i<S.length();i++)
        {
            for(int j=i+1;j<S.length();j++)
            {
                int ucase=0,lcase=0;
                for(int k=i;k<=j;k++)
                {
                    if(Character.isUpperCase(S.charAt(k)))
                    {
                        ucase++;
                    }
                    else
                    {
                        lcase++;
                    }
                }
                if(ucase==lcase) ans++;
            }
        }
        return ans;
    }